packages/css/badge.css

```
@@ -13,135 +13,121 @@
+  }
 
-.ds-badge--success {
-  --dsc-badge-background: var(--ds-color-success-base-default);
-  --dsc-badge-color: var(--ds-color-success-contrast-default);
-}
+  &[data-color='neutral'] {
+    --dsc-badge-background: var(--ds-color-neutral-surface-default);
+    --dsc-badge-color: var(--ds-color-neutral-text-default);
+  }
 
-.ds-badge--warning {
-  --dsc-badge-background: var(--ds-color-warning-base-default);
-  --dsc-badge-color: var(--ds-color-warning-contrast-default);
-}
+  /* If placement is present, we are floating */
+  &[data-placement] {
+    transform-origin: center;
+    position: absolute;
+  }
 
-.ds-badge--danger {
-  --dsc-badge-background: var(--ds-color-danger-base-default);
-  --dsc-badge-color: var(--ds-color-danger-contrast-default);
-}
+  &[data-placement='top-right'][data-overlap='rectangle'] {
+    top: 0;
+    right: 0;
+    transform: translateX(50%) translateY(-50%);
+  }
 
-.ds-badge--neutral {
-  --dsc-badge-background: var(--ds-color-neutral-surface-default);
-  --dsc-badge-color: var(--ds-color-neutral-text-default);
-}
+  &[data-placement='top-left'][data-overlap='rectangle'] {
+    top: 0;
+    left: 0;
+    transform: translateX(-50%) translateY(-50%);
+  }
 
-.ds-badge--sm {
-  --dsc-badge-height: var(--ds-sizing-3);
-  --dsc-badge-width: var(--ds-sizing-3);
+  &[data-placement='bottom-right'][data-overlap='rectangle'] {
+    bottom: 0;
+    right: 0;
+    transform: translateX(50%) translateY(50%);
+  }
 
-  &.ds-badge--count {
-    --dsc-badge-height: var(--ds-sizing-5);
-    --dsc-badge-width: auto;
-    --dsc-badge-padding: 0 var(--ds-spacing-1);
+  &[data-placement='bottom-left'][data-overlap='rectangle'] {
+    bottom: 0;
+    left: 0;
+    transform: translateX(-50%) translateY(50%);
   }
-}
 
-.ds-badge--md {
-  --dsc-badge-height: calc(var(--ds-sizing-3) + calc(var(--ds-spacing-1) / 2));
-  --dsc-badge-width: calc(var(--ds-sizing-3) + calc(var(--ds-spacing-1) / 2));
+  &[data-placement='top-right'][data-overlap='circle'] {
+    top: 14%;
+    right: 14%;
+    transform: translateX(50%) translateY(-50%);
+  }
 
-  &.ds-badge--count {
-    --dsc-badge-height: var(--ds-sizing-6);
-    --dsc-badge-width: auto;
-    --dsc-badge-padding: 0 calc(var(--ds-spacing-1) + calc(var(--ds-spacing-1) / 2));
+  &[data-placement='top-left'][data-overlap='circle'] {
+    top: 14%;
+    left: 14%;
+    transform: translateX(-50%) translateY(-50%);
   }
-}
 
-.ds-badge--lg {
-  --dsc-badge-height: var(--ds-sizing-4);
-  --dsc-badge-width: var(--ds-sizing-4);
+  &[data-placement='bottom-right'][data-overlap='circle'] {
+    bottom: 14%;
+    right: 14%;
+    transform: translateX(50%) translateY(50%);
+  }
 
-  &.ds-badge--count {
-    --dsc-badge-height: var(--ds-sizing-7);
-    --dsc-badge-width: auto;
-    --dsc-badge-padding: 0 var(--ds-spacing-2);
+  &[data-placement='bottom-left'][data-overlap='circle'] {
+    bottom: 14%;
+    left: 14%;
+    transform: translateX(-50%) translateY(50%);
```

Should we use the new CSS syntax? ☺️ 
```suggestion
    translate: -50% 50%;
```

---

PS: by always using left+top (instead of right+bottom, i.e. `right: 0` === `left: 100%`) we can simplify and _always_ use `translate: -50% -50%` instead of changing this property for each placement :)

---

I feel it's more readable to explicitly write X/Y. Is there any benefits do doing it shorthand?

---

> PS: by always using left+top (instead of right+bottom, i.e. `right: 0` === `left: 100%`) we can simplify and _always_ use `translate: -50% -50%` instead of changing this property for each placement :)

don't have anything against this, except that doing the actual two sides for positioning feels better to me..? :P 
So when the placement is `top-left`, we actually change `top` and `left`

---

> I feel it's more readable to explicitly write X/Y. Is there any benefits do doing it shorthand?

By using `translate` directly, we enable using `rotate`, `scale`, `transform` independently, while using `transform` groups all these properties in one declaration, making it harder to do separate transforms. `translate` is actually the full version while `transform` is more of a shorthand as it nulls out the others (and can cause unwanted issues if not being aware of value order). In this particular scenario it does not make much difference, but my experience it is just better to use individual transform properties as a default ☺️ 
More info: https://web.dev/articles/css-individual-transform-properties

---

> > PS: by always using left+top (instead of right+bottom, i.e. `right: 0` === `left: 100%`) we can simplify and _always_ use `translate: -50% -50%` instead of changing this property for each placement :)
> 
> don't have anything against this, except that doing the actual two sides for positioning feels better to me..? :P So when the placement is `top-left`, we actually change `top` and `left`

No strong feelings, just a suggestion to simplify by not needing to change `translate` ☺️